### ISSUE TITLE
Print of Wide, Sortable Horizontal Table

### ISSUE THREAD

**Issue by SarahD** [wenzhixin/bootstrap-table#848]
Hi guys,
I have a super wide, sortable horizontal table (18 columns) and I need to create a print version of it where it shows all of the rows (turns off pagination) and all can be fit (even if tiny) in a landscape view. I would love any thoughts you have on this! Thank you!

http://jsfiddle.net/webdevsarah/pg3pLtdr/9/

All the best,
Sarah

**Comment by PlippiePlop** [comment]
Why don't you use the export extension?
https://github.com/hhurz/tableExport.jquery.plugin

**Comment by SarahD** [comment]
We are, but 9 times out of 10 our clients will just print as opposed to export. :/

**Comment by dabros** [comment]
@SarahD - i dont have a complete solution for you, and ive never worked with export extension so cant directly comment on that, but some points to consider may be:

**1. Printable CSS media**

Its hard to get cross compatible, but if you look around you should be able to find way to make print version styled different, which you maybe can use to hide all unneeded content and then use ".table thead th" and similar to change table styling...

http://stackoverflow.com/questions/1960939/disabling-browser-print-options-headers-footers-margins-from-page

**2. Overwrite print command **

This was my first thought, but seems maybe difficult to truly stop them (ctrl+P is easy, print from file dropdown menu) but if you google and research a little you can maybe figure it out. The goal would be to ideally call your own commands before proceeding to normal print screen, so you can manipulate any styling needed through js.

http://stackoverflow.com/questions/3619609/check-for-when-a-user-has-selected-to-print-using-javascript

**3. Create custom print command**

Much simpler than other solutions i think, though prob contains much of the same css and/or js. You would create very obvious print button, which would link to either export command or some other custom code. You could also combine this with one or both other solutions, to link other print commands to the same function and/or mimic styling set in media inside this custom function and/or overwrite default print then just call that using obvious button.

https://drupaltroll.wordpress.com/2012/03/01/custom-print-function-using-javascript/


**In conclusion**

Many options, mostly pointing to same key css and js code - bit contextual though so cant provide code example of complete solution but shouldnt be too hard if you research and play with it a little : )

**Comment by djhvscf** [comment]
Thanks for you answer @dabros .. Does this work for you, @SarahD ??

**Comment by PlippiePlop** [comment]
Why don't you add a print button in the table toolbar. I think users will click on this. Then add export function to that to print a pdf.

**Comment by wenzhixin** [comment]
You can try the solution what @dabros said, close the issue.
